plugins/net_plugin/net_plugin.cpp

```diff
@@ -1311,6 +1311,11 @@ namespace eosio {
                     uint32_t message_length;
                     auto index = conn->pending_message_buffer.read_index();
                     conn->pending_message_buffer.peek(&message_length, sizeof(message_length), index);
+                    if(message_length > def_send_buffer_size*2) {
```

Comment:
instead of `def_send_buffer_size*2` can we use `app().find_plugin<chain_plugin>()->chain().get_global_properties().max_blk_size * SAFETY_FACTOR` (or some terse version thereof)

A block should be the largest message we receive, and this would make this tolerance "elastic" based on the consensus of producing nodes.

Once voting is enabled, or if we wanted to up the block size and redeploy the test net in the future this would "automatically adjust" as opposed to needing manual tweaks

Reply:
I should say, I'm fine with the fix as is.

Reply:
Agreed, but def_send_buffer_size is similarly disjoint from the max_blk_size. Since the RX buffer is unbounded, all that's important today is that I can atleast receive as much as I can transmit.